bakery/src/scripts/halp.py

```diff
@@ -0,0 +1,40 @@
+from cnxepub.models import TRANSLUCENT_BINDER_ID, TranslucentBinder
+from cnxcommon.urlslug import generate_slug
+
+# Based upon amend_tree_with_slugs from cnx-publishing
+# (https://github.com/openstax/cnx-publishing/blob/master/cnxpublishing/utils.py#L64)
+def amend_tree_with_slugs(tree, title_seq=[]):
+    """Recursively walk through tree and add slug fields"""
+    title_seq = title_seq + [tree['title']]
+    tree['slug'] = generate_slug(*title_seq)
+    if 'contents' in tree:
+        for node in tree['contents']:
+            amend_tree_with_slugs(node, title_seq)
+
+# Based upon model_to_tree from cnx-epub
+# (https://github.com/openstax/cnx-epub/blob/master/cnxepub/models.py#L108)
+def model_to_tree(model, title=None, lucent_id=TRANSLUCENT_BINDER_ID):
+    """Given an model, build the tree::
+        {'id': <id>|'subcol', 'title': <title>, 'contents': [<tree>, ...]}
+    """
+    id = model.ident_hash
+    if id is None and isinstance(model, TranslucentBinder):
+        id = lucent_id
+    md = model.metadata
+    title = title is not None and title or md.get('title')
```

Comment:
Given how we're calling this in our bakery scripts where a title isn't passed, won't `title` and `clean_title` always be the same? Or is this simply because there's something downstream that's specifically expecting `clean_title`?

Reply:
no the title won't be the same - 
```
{'legacy_id': 'col11406',
 'legacy_version': '1.17',
 'license': {'url': 'http://creativecommons.org/licenses/by/4.0/'},
 'title': 'College Physics',
 'tree': {'clean_title': 'College Physics',
          'contents': [{'clean_title': '1 Introduction: The Nature of Science '
                                       'and Physics',
                        'contents': [{'clean_title': 'Introduction to Science '
                                                     'and the Realm of '
                                                     'Physics, Physical '
                                                     'Quantities, and Units',
                                      'id': 'm42119@1.6',
                                      'slug': '1-introduction-to-science-and-the-realm-of-physics-physical-quantities-and-units',
                                      'title': '<span data-type="" itemprop="" '
                                               'class="os-text">Introduction '
                                               'to Science and the Realm of '
                                               'Physics, Physical Quantities, '
                                               'and Units</span>'},
                                     {'clean_title': 'Physics: An Introduction',
                                      'id': 'm42092@1.10',
                                      'slug': '1-1-physics-an-introduction',
                                      'title': '<span '
                                               'class="os-number">1.1</span><span '
                                               'class="os-divider"> '
                                               '</span><span data-type="" '
                                               'itemprop="" '
                                               'class="os-text">Physics: An '
                                               'Introduction</span>'}],
                        'id': '5ae762fc-302a-4170-89fd-6092688fbb58@1.17',
                        'slug': '1-introduction-the-nature-of-science-and-physics',
                        'title': '<span class="os-number">1</span><span '
                                 'class="os-divider"> </span><span '
                                 'data-type="" itemprop="" '
                                 'class="os-text">Introduction: The Nature of '
                                 'Science and Physics</span>'}],
          'id': 'col11406@1.17',
          'slug': 'college-physics',
          'title': 'College Physics'}}
```

`title` is used to generate the slug in disassemble'
'clean_title` is the title of without the html tags

Nothing downstream is expecting `clean_title`, but I thought it would be helpful but at the same time, very preemptive of me.

I'm unsure what is needed and didn't want to break the slug generation, if we use the clean title it breaks the test for disassemble script.

Reply:
I will remove clean_title hahah

Reply:
Whoops, I completely missed the recursive part below earlier. I see / get it now .